crates/agent/src/publications/specs.rs

```diff
@@ -350,6 +350,71 @@ pub fn validate_transition(
     errors
 }
 
+pub async fn enforce_resource_quotas(
```

Comment:
Can you add test coverage for this, akin to how directives are tested? Should be pretty feasible to do since these checks occur so early in the publication process.

Reply:
Done!